pkg/params/validators.go

```diff
@@ -23,20 +23,56 @@ import (
 type TypeHint string
 
 const (
+	TypeBool   TypeHint = "bool"
 	TypeString TypeHint = "string"
 	TypeInt    TypeHint = "int"
+	TypeInt8   TypeHint = "int8"
+	TypeInt16  TypeHint = "int16"
+	TypeInt32  TypeHint = "int32"
+	TypeInt64  TypeHint = "int64"
 	TypeUint   TypeHint = "uint"
-	TypeBool   TypeHint = "bool"
+	TypeUint8  TypeHint = "uint8"
+	TypeUint16 TypeHint = "uint16"
+	TypeUint32 TypeHint = "uint32"
+	TypeUint64 TypeHint = "uint64"
 )
 
 var typeHintValidators = map[TypeHint]ParamValidator{
-	TypeInt:  ValidateNumber, // TODO: more specific
-	TypeUint: ValidateNumber, // TODO: more specific
-	TypeBool: ValidateBool,
+	TypeBool:   ValidateBool,
+	TypeInt:    ValidateInt(strconv.IntSize),
```

Comment:
I'm not totally sure whether we should use `IntSize` here - maybe it's better to stick with a hardcoded 64 in that case. Reason being: if used in a 32 bit kubectl-gadget binary, this would be different to what runs on the server and be too strict. It's a bit edge casey, I admit ;)

Reply:
I had a offline talk with @flyth, the plan is to remove the TypeInt / TypeUint as those types with a specific size are difficult to handle in distributed systems as client and server could be compiled on different architectures. I'll merge this PR and those will be removed in https://github.com/inspektor-gadget/inspektor-gadget/pull/1281.